packages/ui/src/TextField.tsx

```diff
@@ -241,121 +215,140 @@ export const TextField = ({
-              flexDirection: "row",
-              justifyContent: "center",
-              alignItems: "center",
-              // height: multiline || grow ? undefined : 40,
-              minHeight: getHeight(),
-              width: "100%",
-              // Add padding so the border doesn't mess up layouts
-              paddingHorizontal: focused ? 10 : 14,
-              paddingVertical: focused ? 0 : 4,
-              borderColor,
-              borderWidth: focused ? 5 : 1,
-              borderRadius: 16,
-              backgroundColor: disabled ? theme.surface.neutral : theme.surface.base,
-              overflow: "hidden",
+              color: theme.text.primary,
+              fontSize: 16,
+              fontWeight: 600,
+              lineHeight: 22.4, // 140% of 16 per design
             }}
-            onPress={() => {
-              // This runs on web
-              onTap();
+          >
+            {title}
+          </Text>
+        )}
+        {state === "error" && errorText && (
+          <View style={{flexDirection: "row", alignItems: "center", marginVertical: 2}}>
+            <Icon color="error" iconName="triangle-exclamation" size="sm" />
+            <View style={{marginLeft: 4}}>
+              <Text style={{color: theme.text.error, fontSize: 12}}>{errorText}</Text>
+            </View>
+          </View>
+        )}
+        <Wrapper
+          style={{
+            flexDirection: "row",
+            alignItems: "center",
+            backgroundColor: state === "disabled" ? theme.surface.neutralLight : theme.surface.base,
+            borderColor,
+            borderWidth: focused ? 3 : 1,
+            paddingHorizontal: focused ? 10 : 12,
+            paddingVertical: focused ? 6 : 8,
+            borderRadius: 4,
+            overflow: "hidden",
+          }}
+        >
+          <TextInput
+            ref={(ref) => {
+              if (inputRef) {
+                inputRef(ref);
+              }
             }}
-            onTouchStart={() => {
-              // This runs on mobile
-              onTap();
+            accessibilityHint="Enter text here"
+            accessibilityLabel="Text input field"
+            autoCapitalize={type === "text" ? "sentences" : "none"}
+            autoCorrect={shouldAutocorrect}
+            blurOnSubmit={blurOnSubmit}
+            editable={isEditable}
+            keyboardType={keyboardType as KeyboardTypeOptions}
+            multiline={multiline}
+            numberOfLines={rows || 4}
+            placeholder={placeholderText}
+            placeholderTextColor={theme.text.secondaryLight}
+            returnKeyType={type === "number" || type === "decimal" ? "done" : returnKeyType}
+            secureTextEntry={type === "password"}
+            style={defaultTextInputStyles}
+            testID={testID}
+            textContentType={textContentType}
+            underlineColorAndroid="transparent"
+            value={displayValue}
+            onBlur={() => {
```

Comment:
will you please add prop `selectable` to TextInput
either 
```
const isSelectable = !disabled
  selectable={isSelectable}
// or
selectable= !disabled
```

This issue-  The user should not be able to select if disabled, we do not want onBlur function to run if element is disabled.

Reply:
Is there a way to simply disable selection on `TextInput` that you know of? That is a react-native component and I didn't see an easy way without manually changing borders, selection characteristics, etc. It is an issue with our current `TextField` as well.